File: scripts/lib/commands/meta.js
const { parseArgs, requirePositional } = require('../args');
const { AppError } = require('../errors');
const { ERROR_CODES } = require('../constants');
const { pickIssueType } = require('../utils');

async function runFields(args, context) {
  const parsed = parseArgs(args, {});
  const projectKey = parsed.options.project;
  const issueTypeInput = parsed.options['issue-type'];

  if (!projectKey) {
    const data = await context.client.getFields();
    return {
      data,
      tableRows: data.map((field) => ({
        id: field.id,
        name: field.name,
        custom: Boolean(field.custom),
        type: field.schema ? field.schema.type : ''
      }))
    };
  }

  const createMeta = await context.client.getCreateMeta(projectKey);
  const project = createMeta.projects && createMeta.projects[0];
  if (!project) {
    throw new AppError(ERROR_CODES.VALIDATION, `Project "${projectKey}" not found in createmeta.`);
  }

  const issueType = issueTypeInput
    ? pickIssueType(project.issuetypes, issueTypeInput)
    : project.issuetypes && project.issuetypes[0];
  if (!issueType) {
    throw new AppError(ERROR_CODES.VALIDATION, `Issue type "${issueTypeInput}" not found in "${projectKey}".`);
  }

  const fields = issueType.fields || {};
  const rows = Object.keys(fields).map((id) => ({
    id,
    name: fields[id].name,
    required: Boolean(fields[id].required),
    hasDefault: fields[id].hasDefaultValue === true
  }));

  return {
    data: {
      project: project.key,
      issueType: issueType.name,
      fields
    },
    tableRows: rows
  };
}

async function runIssueTypes(args, context) {
  const parsed = parseArgs(args, {});
  const projectKey = parsed.options.project || requirePositional(parsed.positionals, 0, 'PROJECT-KEY');
  const createMeta = await context.client.getCreateMeta(projectKey);
  const project = createMeta.projects && createMeta.projects[0];
  if (!project) {
    throw new AppError(ERROR_CODES.VALIDATION, `Project "${projectKey}" not found.`);
  }
  const issuetypes = project.issuetypes || [];
  return {
    data: {
      project: project.key,
      issueTypes: issuetypes
    },
    tableRows: issuetypes.map((type) => ({
      id: type.id,
      name: type.name,
      subtask: Boolean(type.subtask)
    }))
  };
}

async function runEditMeta(args, context) {
  const parsed = parseArgs(args, {});
  const issueKey = requirePositional(parsed.positionals, 0, 'ISSUE-KEY');
  const data = await context.client.getEditMeta(issueKey);
  const fields = data.fields || {};
  return {
    data,
    tableRows: Object.keys(fields).map((id) => ({
      id,
      name: fields[id].name,
      required: Boolean(fields[id].required)
    }))
  };
}

const subcommands = new Map([
  ['fields', runFields],
  ['issue-types', runIssueTypes],
  ['editmeta', runEditMeta]
]);

async function run(args, context) {
  const [subcommand, ...rest] = args;
  if (!subcommand || !subcommands.has(subcommand)) {
    throw new AppError(ERROR_CODES.VALIDATION, 'meta requires subcommand fields|issue-types|editmeta.');
  }
  return subcommands.get(subcommand)(rest, context);
}

module.exports = {
  name: 'meta',
  description: 'Metadata operations (fields/issue-types/editmeta)',
  run
};

File: scripts/python/lib/commands/meta.py
from ..args import parse_args, require_positional
from ..errors import AppError
from ..constants import ERROR_CODES
from ..utils import pick_issue_type

NAME = 'meta'

def run_fields(args, context):
    parsed = parse_args(args)
    project_key = parsed['options'].get('project')
    issue_type_input = parsed['options'].get('issue-type')

    if not project_key:
        data = context['client'].get_fields()
        return {
            'data': data,
            'tableRows': [
                {
                    'id': field.get('id'),
                    'name': field.get('name'),
                    'custom': bool(field.get('custom')),
                    'type': field.get('schema', {}).get('type') if field.get('schema') else ''
                } for field in data
            ]
        }

    create_meta = context['client'].get_create_meta(project_key)
    projects = create_meta.get('projects') or []
    project = projects[0] if projects else None
    
    if not project:
        raise AppError(ERROR_CODES.VALIDATION, f"Project \"{project_key}\" not found in createmeta.")

    issuetypes = project.get('issuetypes') or []
    issue_type = pick_issue_type(issuetypes, issue_type_input) if issue_type_input else (issuetypes[0] if issuetypes else None)
    
    if not issue_type:
        raise AppError(ERROR_CODES.VALIDATION, f"Issue type \"{issue_type_input}\" not found in \"{project_key}\".")

    fields = issue_type.get('fields') or {}
    rows = [
        {
            'id': field_id,
            'name': fields[field_id].get('name'),
            'required': bool(fields[field_id].get('required')),
            'hasDefault': fields[field_id].get('hasDefaultValue') is True
        } for field_id in fields
    ]

    return {
        'data': {
            'project': project.get('key'),
            'issueType': issue_type.get('name'),
            'fields': fields
        },
        'tableRows': rows
    }

def run_issue_types(args, context):
    parsed = parse_args(args)
    project_key = parsed['options'].get('project') or require_positional(parsed['positionals'], 0, 'PROJECT-KEY')
    
    create_meta = context['client'].get_create_meta(project_key)
    projects = create_meta.get('projects') or []
    project = projects[0] if projects else None
    
    if not project:
        raise AppError(ERROR_CODES.VALIDATION, f"Project \"{project_key}\" not found.")
        
    issuetypes = project.get('issuetypes') or []
    return {
        'data': {
            'project': project.get('key'),
            'issueTypes': issuetypes
        },
        'tableRows': [
            {
                'id': t.get('id'),
                'name': t.get('name'),
                'subtask': bool(t.get('subtask'))
            } for t in issuetypes
        ]
    }

def run_edit_meta(args, context):
    parsed = parse_args(args)
    issue_key = require_positional(parsed['positionals'], 0, 'ISSUE-KEY')
    data = context['client'].get_edit_meta(issue_key)
    
    fields = data.get('fields') or {}
    return {
        'data': data,
        'tableRows': [
            {
                'id': field_id,
                'name': fields[field_id].get('name'),
                'required': bool(fields[field_id].get('required'))
            } for field_id in fields
        ]
    }

SUBCOMMANDS = {
    'fields': run_fields,
    'issue-types': run_issue_types,
    'editmeta': run_edit_meta
}

def run(args, context):
    if not args:
        raise AppError(ERROR_CODES.VALIDATION, 'meta requires subcommand fields|issue-types|editmeta.')
    
    subcommand = args[0]
    rest = args[1:]
    
    if subcommand not in SUBCOMMANDS:
        raise AppError(ERROR_CODES.VALIDATION, 'meta requires subcommand fields|issue-types|editmeta.')
    
    return SUBCOMMANDS[subcommand](rest, context)
